### PR TITLE
Feat: Improvements in helper classes

### PR DESCRIPTION
**Pull request by 2002Bishwajeet**
## What does this PR do?

Make Constructors private of the following classes:

- Permission
- Query
- ID
- Role

 This was done to improve the IntelliSense ( earlier it used to suggest the constructor though all the methods are static) 

## Test Plan

Manual QA

 
 ### Constructor Public
 
image

### When Constructor Private

image

## Related PRs and Issues

Nil

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Always😉

### CODE REVIEW

File: templates/dart/lib/id.dart.twig [hunk 1]
```diff
@@ -1,6 +1,8 @@
 part of {{ language.params.packageName }};
 
 class ID {
+    const ID._();
```
[the remark this answers is not in the document]
Comment (2002Bishwajeet) [reply templates/dart/lib/id.dart.twig:4]: Honestly, nothing! Do you want me to omit that? It's gonna be the same regardless of whether they remain `const` or not

[the remark this answers is not in the document]
Comment (stnguyen90) [reply templates/dart/lib/id.dart.twig:4]: Probably best to leave out anything unnecessary.

**Review by stnguyen90 — APPROVED** [review]
LGTM!

**Review by lohanidamodar — APPROVED** [review]
LTGM.

**Pull request merged**